### ISSUE TITLE
[BUG/FR] Alias Not Suggested If Filename's Prefix  Like The Alias

### ISSUE THREAD

**Issue by almogtzabari** [tadashi-aikawa/obsidian-various-complements-plugin#73]
Hi,
First, I would like to thank you for implementing [this](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/issues/64) awesome feature!
Now, I think that I found a bug regarding this feature (not sure if its really a bug though). I'll try to explain with an example.

### Current Behavior
If we have a file with the name `John Gear`, and that file has an alias `John`, then when starting to type `John` it suggests `[[John Gear]]`, but doesn't suggest `[[John Gear|John]]` at all.

### Desired Behavior
When starting to type `John` the plugin will suggest both `[[John Gear|John]]` and `[[John Gear]]`. 
Personally, I prefer that the alias `[[John Gear|John]]` would be suggested as the first choice (higher priority) because we can always keep typing `johng` until it suggests `[[John Gear]]`.

Does it make sense to you as it does to me?
Thanks in advance :)

**Comment by tadashi-aikawa** [comment]
Hi, @almogtzabari. Thank you for your report :)

I could reproduce it, and it is a bit of a difficult problem to resolve. However, I agree with you perfectly, although I don't use aliases as the inserted text.

It may take some time, but I'll try to resolve this issue. Kind regards.

**Comment by almogtzabari** [comment]
Ok, thank you very much :)

**Comment by tadashi-aikawa** [comment]
@almogtzabari 
I have updated beta version to [obsidian-various-complements-plugin-beta v5.1.0-beta1](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin-beta/releases/tag/5.1.0.beta1) 🚀 

If you use [BRAT](https://github.com/TfTHacker/obsidian42-brat), please confirm whether it works as you expected. 

If you don't use it, please verify the below GIF.

![](https://user-images.githubusercontent.com/9500018/152639803-d7e02988-f0f9-41c3-a20a-93e44dd17afd.gif)

**Comment by almogtzabari** [comment]
> @almogtzabari I have updated beta version to [obsidian-various-complements-plugin-beta v5.1.0-beta1](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin-beta/releases/tag/5.1.0.beta1) 🚀
> 
> If you use [BRAT](https://github.com/TfTHacker/obsidian42-brat), please confirm whether it works as you expected.
> 
> If you don't use it, please verify the below GIF.
> 
> ![](https://user-images.githubusercontent.com/9500018/152639803-d7e02988-f0f9-41c3-a20a-93e44dd17afd.gif)


I don't use BRAT, but the behavior shown in your GIF is exactly what I wished for!

You are awesome! Thank you very much!!!

**Comment by tadashi-aikawa** [comment]
Thank you for your quick confirmation :)

I will release v5.1.0 officially in a few days because I updated a few code bases that might make any bugs 👍

**Comment by almogtzabari** [comment]
@tadashi-aikawa 
I wasn't aware of BRAT, but I downloaded it now.
I followed BRAT's instructions about how to add a new plugin (I added the plugin's Github address to BRAT, hit refresh on the plugins list) but It doesn't install the beta version, only the 5.0.1 .
Is there another link I should use? (I used `https://github.com/tadashi-aikawa/obsidian-various-complements-plugin`)

**Comment by tadashi-aikawa** [comment]
@almogtzabari 
Please use `https://github.com/tadashi-aikawa/obsidian-various-complements-plugin-beta`. It is only used for a beta test.

```diff
- https://github.com/tadashi-aikawa/obsidian-various-complements-plugin
+ https://github.com/tadashi-aikawa/obsidian-various-complements-plugin-beta

```

**Comment by almogtzabari** [comment]
> @almogtzabari Please use `https://github.com/tadashi-aikawa/obsidian-various-complements-plugin-beta`. It is only used for a beta test.
> 
> ```diff
> - https://github.com/tadashi-aikawa/obsidian-various-complements-plugin
> + https://github.com/tadashi-aikawa/obsidian-various-complements-plugin-beta
> ```

Right :)

I installed the beta version, and your changes work like a charm! Amazing work!
BTW, I like the small icons near each suggestion!

**Comment by tadashi-aikawa** [comment]
That's good! And glad to hear that you like the small icons :)

**Comment by tadashi-aikawa** [comment]
Released at v5.1.0 🚀 

https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/releases/tag/5.1.0
